src/main/java/uk/gov/hmcts/darts/common/datamanagement/api/DataManagementFacadeImpl.java

```diff
@@ -1,125 +1,173 @@
 package uk.gov.hmcts.darts.common.datamanagement.api;
 
+import com.azure.storage.blob.models.BlobStorageException;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
 import org.springframework.stereotype.Service;
-import uk.gov.hmcts.darts.common.datamanagement.component.impl.DownloadableExternalObjectDirectories;
+import uk.gov.hmcts.darts.common.datamanagement.component.impl.DownloadResponseMetaData;
 import uk.gov.hmcts.darts.common.datamanagement.enums.DatastoreContainerType;
+import uk.gov.hmcts.darts.common.datamanagement.helper.StorageOrderHelper;
+import uk.gov.hmcts.darts.common.entity.AnnotationDocumentEntity;
 import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectRecordStatusEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
 import uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
+import uk.gov.hmcts.darts.datamanagement.exception.FileNotDownloadedException;
 import uk.gov.hmcts.darts.dets.config.DetsDataManagementConfiguration;
 
 import java.io.UncheckedIOException;
-import java.util.ArrayList;
+import java.text.MessageFormat;
 import java.util.List;
 import java.util.Optional;
 
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+
 @Slf4j
 @Service
+@RequiredArgsConstructor
 public class DataManagementFacadeImpl implements DataManagementFacade {
 
     private final List<BlobContainerDownloadable> supportedDownloadableContainers;
-
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final StorageOrderHelper storageOrderHelper;
     private final DetsDataManagementConfiguration configuration;
```

Comment:
I take it this `DetsDataManagementConfiguration` dependency here can be removed now?